src/main/g8/src/main/scala/worker/Worker.scala

```diff
@@ -1,37 +1,41 @@
+/**
+ * Copyright (C) 2009-2017 Lightbend Inc. <http://www.lightbend.com>
+ */
 package worker
 
 import java.util.UUID
 
 import akka.actor.SupervisorStrategy.{Restart, Stop}
 import akka.actor._
-import akka.cluster.singleton.{ClusterSingletonProxy, ClusterSingletonProxySettings}
 
 import scala.concurrent.duration._
 
+/**
+ * The worker is actually more of a middle manager, delegating the actual work
+ * to the WorkExecutor, supervising it and keeping itself available to interact with the work master.
+ */
 object Worker {
 
-  def props(workExecutorProps: Props, registerInterval: FiniteDuration = 10.seconds): Props =
-    Props(new Worker(workExecutorProps, registerInterval))
+  def props(): Props = Props(new Worker())
 
-  case class WorkComplete(result: Any)
 }
 
-class Worker(workExecutorProps: Props, registerInterval: FiniteDuration)
+class Worker()
   extends Actor with ActorLogging {
   import MasterWorkerProtocol._
-  import Worker._
+  import context.dispatcher
 
-  val workerId = UUID.randomUUID().toString
 
-  import context.dispatcher
+  val workerId = UUID.randomUUID().toString
+  val registerInterval = context.system.settings.config.getDuration("distributed-workers.worker-registration-interval").getSeconds.seconds
 
   val masterProxy = context.actorOf(
     MasterSingleton.proxyProps(context.system),
     name = "masterProxy")
```

Comment:
better to create the proxy outside the Worker actor and pass it as param, then same proxy can be shared among all workers (on a node)